### PR TITLE
Improving cluster quickstart docs

### LINKED ISSUE #311: Don't use --background flag with docker
From Gitter:

Michael Hausenblas @mhausenblas
yo, folks, anyone an idea why docker run cockroachdb/cockroach:beta-20160519 start --background is dying on me?
FWIW, I experience the same when I build it from scratch like so:

```
 FROM ubuntu:14.04
MAINTAINER Michael Hausenblas "michael.hausenblas@gmail.com"
ENV REFRESHED_AT 2016-05-22T07:37Z

RUN    apt-get -qq update \ 
    && apt-get -f install \
    && apt-get -y install wget \
    && wget https://binaries.cockroachdb.com/cockroach-beta-20160519.linux-amd64.tgz \
    && tar xfz cockroach-beta-20160519.linux-amd64.tgz

EXPOSE 8080
ENTRYPOINT ["cockroach-beta-20160519.linux-amd64/cockroach"]
CMD ["start", "--background"]
```

Funnily enough, when I start out with a ubuntu:14.04 image and do the steps manually, it works 8->

Ben Darnell @bdarnell 05:56
@mhausenblas Don’t use —background when running under docker (or systemd, or anything else that’s going to monitor the process). —background is just for starting from a plain shell

Michael Hausenblas @mhausenblas 05:56
got it, thanks @bdarnell!
might wanna make this explicit in the docs? also, adding the Dockerfile in https://hub.docker.com/r/cockroachdb/cockroach/ might help
or did I fail RTFMing? 

Ben Darnell @bdarnell 06:02
no, we don’t have good docs for running in docker right now. we still need to sort out the best way to manage volumes, ports, etc
we recommend running without docker for now
also, i understand that we need to document that you should run docker run cockroachdb/cockroach start instead of start —background, but how does putting the Dockerfile on hub.docker.com help? the Dockerfile is all implementation details, not instructions

### LINKED ISSUE #317: Emphasize use of --insecure for any but localhost
@bdarnell: Emphasize that for anything but localhost, you need to use either the `—insecure` flag or all the certificate flags.

### PR DESCRIPTION
**Pull request by jseldess**
1. Updated `install-cockroachdb.md` to suggest adding the binary directory to the `PATH`. Therefore, also updated all instances of `./cockroach` to `cockroach`.
2. Revised binary quickstart (`start-a-local-cluster.md`) so commands are easily accessible while details are no longer hidden. Fixes #317 
3. Added quickstart instructions for running a local cluster in Docker (`start-a-local-cluster-in-docker.md`). Also added `From Binary` and `In Docker` toggles to `start-a-local-cluster.md` and `start-a-local-cluster-in-docker.md` so they seem like one page, whereas they're really two. This is good for pageview tracking. Fixes #311 

HTML versions available here:
- http://cockroach-draft-docs.s3-website-us-east-1.amazonaws.com/install-cockroachdb.html
- http://cockroach-draft-docs.s3-website-us-east-1.amazonaws.com/start-a-local-cluster.html
- http://cockroach-draft-docs.s3-website-us-east-1.amazonaws.com/start-a-local-cluster-in-docker.html

Will create a Docker-specific version of `secure-a-cluster.md` in a separate PR.

@bdarnell, @petermattis, @mberhault, PTAL.

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/389)

### CODE REVIEW

**Comment by jseldess** [comment]
@bdarnell, @petermattis, @mberhault, Sorry about the unruly list of commits. Best if you don't review them one-by-one, as some ended up irrelevant.

**Comment by bdarnell** [comment]
:lgtm:

---

Review status: 0 of 19 files reviewed at latest revision, 2 unresolved discussions.

---

*[install-cockroachdb.md, line 188 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/389#-KKPGRz_g2xZcQt1zqxI:-KKPGRzaWA2b7vKc22Ue:-954976095) ([raw file](https://github.com/cockroachdb/docs/blob/b6d039dd25d90772d12750b8ff603afd69c074e4/install-cockroachdb.md#L188)):*
> ```Markdown
>     <p>Compile the CockroachDB binary:</p>
> 
>     <div class="highlighter-rouge"><pre class="highlight"><code data-eventcategory="mac-source-step3"><span class="gp" data-eventcategory="mac-source-step3">$ </span><span class="nb">cd</span> <span class="nv">$GOPATH</span>/src/github.com/cockroachdb/cockroach<br><span class="gp" data-eventcategory="mac-source-step3">$ </span>git checkout {{site.data.strings.version}}<br><span class="gp">$ </span>make build</code></pre></div>
> ```

Now that we're recommending putting the binary on your path, this should probably use `make install` instead of `make build`. Then the new bullet would be something like "`make install` puts the binary in `$GOPATH/bin`. Add this directory to your `PATH` if it isn't already there"

---

*[install-cockroachdb.md, line 339 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/389#-KKPH-hAS4vahIi9U7L-:-KKPH-hAS4vahIi9U7L0:-1914926843) ([raw file](https://github.com/cockroachdb/docs/blob/b6d039dd25d90772d12750b8ff603afd69c074e4/install-cockroachdb.md#L339)):*
> ```Markdown
>   </li>
>   <li>
>     <p>Add the directory containing the binary to your <code>PATH</code>. This makes it easy to execute <a href="cockroach-commands.html">cockroach commands</a> from any shell.</p>
> ```

Same here (`make install`)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/389#-:-KKPJiG3y0Tko32erRRt:1417088721)*

File: manual-deployment.md [hunk 1]
```diff
@@ -171,7 +171,7 @@ For more information, see [Configure Replication Zones](configure-replication-zo
 You can run the built-in SQL client from any machine with the `cockroach` binary, CA cert, client certificate, and client key. Make sure the machine you want to use has these files and then start the client:  
 
 ~~~ shell
-$ ./cockroach sql --url=postgresql://root@<node1-hostname>:26257/?sslcert=root.cert&sslkey=root.key&sslmode=verify-full&sslrootcert=ca.cert
+$ cockroach sql --url=postgresql://root@<node1-hostname>:26257/?sslcert=root.cert&sslkey=root.key&sslmode=verify-full&sslrootcert=ca.cert
```
Comment (mberhault) [inline manual-deployment.md]: just noticed this now: the url needs to be quoted, or the shell will barf on `&`.

File: use-the-built-in-sql-client.md [hunk 2]
```diff
@@ -64,20 +64,20 @@ In these examples, we connect a SQL shell to a **secure cluster**.
 
 ~~~ shell
 # Using standard connection flags:
-$ ./cockroach sql --ca-cert=certs/ca.cert --cert=certs/maxroach.cert --key=certs/maxroach.key --user=maxroach --host=roachcluster.com --port=26257 --database=critterdb 
+$ cockroach sql --ca-cert=certs/ca.cert --cert=certs/maxroach.cert --key=certs/maxroach.key --user=maxroach --host=roachcluster.com --port=26257 --database=critterdb 
 
 # Using the --url flag:
-$ ./cockroach sql --url=postgresql://maxroach@roachcluster.com:26257/critterdb?sslcert=certs/maxroach.crt&sslkey=certs/maxroach.key&sslmode=verify-full&sslrootcert=certs/ca.crt 
+$ cockroach sql --url=postgresql://maxroach@roachcluster.com:26257/critterdb?sslcert=certs/maxroach.crt&sslkey=certs/maxroach.key&sslmode=verify-full&sslrootcert=certs/ca.crt
```
Comment (mberhault) [inline use-the-built-in-sql-client.md]: same thing throughout this file. Any use of `&` on the command line needs to be quoted. eg:
```
$ cockroach sql --url="postgresql://maxroach@roachcluster.com:26257/critterdb?sslcert=certs/maxroach.crt&sslkey=certs/maxroach.key&sslmode=verify-full&sslrootcert=certs/ca.crt"
```

**Comment by mberhault** [comment]
LGTM modulo Ben's comment and the `&` escaping (sorry I missed it earlier)

**Comment by jseldess** [comment]
Review status: 0 of 19 files reviewed at latest revision, 4 unresolved discussions.

---

*[install-cockroachdb.md, line 188 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/389#-KKPGRz_g2xZcQt1zqxI:-KKU2bbPBOl6XWqhFpxV:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/b6d039dd25d90772d12750b8ff603afd69c074e4/install-cockroachdb.md#L188)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary>
> Now that we're recommending putting the binary on your path, this should probably use `make install` instead of `make build`. Then the new bullet would be something like "`make install` puts the binary in `$GOPATH/bin`. Add this directory to your `PATH` if it isn't already there"
</details>
Done.

---

*[install-cockroachdb.md, line 339 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/389#-KKPH-hAS4vahIi9U7L-:-KKU2aXo3Qow8BZAOdWs:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/b6d039dd25d90772d12750b8ff603afd69c074e4/install-cockroachdb.md#L339)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary>
> Same here (`make install`)
</details>
Done.

---

*[manual-deployment.md, line 174 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/389#-KKM5xSS0484ox4CZlEI-r1-174:-KKU5SqkuHWNHoAAHl_l:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/b6d039dd25d90772d12750b8ff603afd69c074e4/manual-deployment.md#L174)):*
<details><summary><i>Previously, mberhault (marc) wrote…</i></summary>
> just noticed this now: the url needs to be quoted, or the shell will barf on `&`.
</details>
Done.

---

*[use-the-built-in-sql-client.md, line 70 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/389#-KKM5xSTrQ_QOf8DjC9T-r1-70:-KKU5SD2JDzrLEhfTkI4:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/b6d039dd25d90772d12750b8ff603afd69c074e4/use-the-built-in-sql-client.md#L70)):*
<details><summary><i>Previously, mberhault (marc) wrote…</i></summary>
> same thing throughout this file. Any use of `&` on the command line needs to be quoted. eg:
> 
> ```
> 
> $ cockroach sql --url="postgresql://maxroach@roachcluster.com:26257/critterdb?sslcert=certs/maxroach.crt&sslkey=certs/maxroach.key&sslmode=verify-full&sslrootcert=certs/ca.crt"
> 
> ```
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/389)*

**Pull request merged**